### DISCUSSION TITLE
GitHub Actions base image macOS-10.14 is missing Xcode 10.* tools

### DISCUSSION THREAD

**Discussion by ppaulweber** [community/community#25811] [Actions]
<p>Dear Github community,</p>
<p>could it be that you’ve changed the OS base image in GitHub Actions of the <strong>macOS-10.14</strong> tag version? Because, my MacOS builds currently are failing due to a not installed <strong>Xcode 10.3</strong> version? When I try to <strong>xcode-select</strong>  and switch du a particular <strong>Xcode</strong> version, the <strong>10.*</strong> version are not present any more. The installed Xcode version is limited to <strong>Xcode 11.*</strong> , which does not conform to the <a href="https://help.github.com/en/github/automating-your-workflow-with-github-actions/software-installed-on-github-hosted-runners#macos-1014" rel="nofollow noopener">documented software tools written in the documentation</a>:</p>
<pre><code>ls -n /Applications/ | grep Xcode*
lrwxr-xr-x 1 0 80 28 Nov 2 09:18 Xcode.app -&gt; /Applications/Xcode_11.1.app
drwxr-xr-x 3 501 20 96 Sep 21 02:01 Xcode_11.1.app
lrwxr-xr-x 1 501 80 28 Nov 2 08:42 Xcode_11.1_beta.app -&gt; /Applications/Xcode_11.1.app
drwxr-xr-x 3 501 20 96 Oct 23 11:29 Xcode_11.2.app
lrwxr-xr-x 1 501 80 28 Nov 2 09:01 Xcode_11.2_beta.app -&gt; /Applications/Xcode_11.2.app
drwxr-xr-x 3 501 20 96 Sep 11 11:50 Xcode_11.app
lrwxr-xr-x 1 501 80 26 Nov 2 08:22 Xcode_11_beta.app -&gt; /Applications/Xcode_11.app
</code></pre>
<p>I’ve observed that the <code>uname</code> of the OS <code>macOS-10.14</code> label changed from: </p>
<pre><code>Darwin 18.7.0 Tue Aug 20 16:57:14 PDT 2019 root:xnu-4903.271.2~2/RELEASE_X86_64
</code></pre>
<p>to:</p>
<pre><code>Darwin 19.0.0 Thu Oct 17 16:17:15 PDT 2019 root:xnu-6153.41.3~29/RELEASE\_X86\_64
</code></pre>
<p>Best regards</p>

**Comment by blane1988** [comment]
<p>I am unable to us Xcode 10.3 starting this morning. I get this error now. Using Xcode 11.1 succeeds on this step however.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/d/d8e186ad3f660bcbb8301a23d764f5607f68bb56.png" data-download-href="/uploads/short-url/uWCjTZo7eXVtyY2llMkapcO7SAu.png?dl=1" title="Screen Shot 2019-11-06 at 10.21.26 AM.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/42759546/181093567-06913933-45a5-412d-9b51-3b70e9dd101e.png" alt="Screen Shot 2019-11-06 at 10.21.26 AM.png" data-base62-sha1="uWCjTZo7eXVtyY2llMkapcO7SAu" width="690" height="165" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/d/d8e186ad3f660bcbb8301a23d764f5607f68bb56_2_690x165.png, https://aws1.discourse-cdn.com/github/optimized/2X/d/d8e186ad3f660bcbb8301a23d764f5607f68bb56_2_1035x247.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/d/d8e186ad3f660bcbb8301a23d764f5607f68bb56_2_1380x330.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/d/d8e186ad3f660bcbb8301a23d764f5607f68bb56_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen Shot 2019-11-06 at 10.21.26 AM.png</span><span class="informations">1606×386 38.3 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by elanarul** [comment]
<p>Seeing same error </p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/0/0ca1cf4be576af3d944b179300484236afb950af.png" data-download-href="/uploads/short-url/1NKp8vFoyeJOp2Tm1PL0K8iizkX.png?dl=1" title="Screen Shot 2019-11-06 at 11.55.34 AM.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/43356076/181093571-b8a37dc3-0ef6-4d56-b3ce-fbf9b544acc0.png" alt="Screen Shot 2019-11-06 at 11.55.34 AM.png" data-base62-sha1="1NKp8vFoyeJOp2Tm1PL0K8iizkX" width="690" height="93" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/0/0ca1cf4be576af3d944b179300484236afb950af_2_690x93.png, https://aws1.discourse-cdn.com/github/original/2X/0/0ca1cf4be576af3d944b179300484236afb950af.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/0/0ca1cf4be576af3d944b179300484236afb950af.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/0/0ca1cf4be576af3d944b179300484236afb950af_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen Shot 2019-11-06 at 11.55.34 AM.png</span><span class="informations">987×134 12.6 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by blane1988** [comment]
<p>When I try to list out contents of the directory, it says it doesn’t exists. It’s like Xcode 10.3 was removed.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/b/b95e5ef0c85d79f13f7476125f581f6eebd8378d.png" data-download-href="/uploads/short-url/qrQwhsAwLThiGYPJAKYxAQpDCnb.png?dl=1" title="Screen Shot 2019-11-06 at 3.07.07 PM.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/42759546/181093575-00c1fefe-0a0b-4bd3-b6b3-6e6228a49a16.png" alt="Screen Shot 2019-11-06 at 3.07.07 PM.png" data-base62-sha1="qrQwhsAwLThiGYPJAKYxAQpDCnb" width="690" height="97" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/b/b95e5ef0c85d79f13f7476125f581f6eebd8378d_2_690x97.png, https://aws1.discourse-cdn.com/github/optimized/2X/b/b95e5ef0c85d79f13f7476125f581f6eebd8378d_2_1035x145.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/b/b95e5ef0c85d79f13f7476125f581f6eebd8378d.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/b/b95e5ef0c85d79f13f7476125f581f6eebd8378d_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen Shot 2019-11-06 at 3.07.07 PM.png</span><span class="informations">1232×174 21.2 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by elanarul** [comment]
<p>I tried to list all the versions of XCode 10 but didn’t work. So using XCode 11</p>

**Comment by norio-nomura** [comment]
<p>GitHub Actions added macOS 10.15 without Xcode 9.x/10.x, and removed macOS 10.14<br>
<a href="https://github.blog/changelog/2019-11-06-github-actions-macos-virtual-environment-updated-to-catalina/" rel="nofollow noopener">https://github.blog/changelog/2019-11-06-github-actions-macos-virtual-environment-updated-to-catalina/</a><br>
Related issue on GitHub Actions Virtual Environments<br>
<a href="https://github.com/actions/virtual-environments/issues/78" rel="nofollow noopener">https://github.com/actions/virtual-environments/issues/78</a></p>

**Comment by BrightRan** [comment, marked as answer]
<p>Hi ppaulweber,</p>
<p>Now the self-hosted runners for GitHub Actions is in beta, you can try to install a self-hosted macOS runner on your local macOS machine (or your macOS VM) with the Xcode 10.* installed as a workaround for now.<br>
More details about Self-hosted runners for GitHub Actions, please reference here: <a href="https://github.blog/2019-11-05-self-hosted-runners-for-github-actions-is-now-in-beta/" rel="nofollow noopener">https://github.blog/2019-11-05-self-hosted-runners-for-github-actions-is-now-in-beta/</a></p>
